sbi/utils/torchutils.py

```diff
@@ -21,34 +21,38 @@ def process_device(device: str) -> str:
     Throws an AssertionError if the prior is not matching the training device not.
     """
 
-    if not device == "cpu":
-        assert device.startswith("cuda"), f"Invalid device string: {device}."
-        try:
-            torch.zeros(1).to(device)
-            warnings.warn(
-                """GPU was selected as a device for training the neural network. Note
-                   that we expect **no** significant speed ups in training for the
-                   default architectures we provide. Using the GPU will be effective
-                   only for large neural networks with operations that are fast on the
-                   GPU, e.g., for a CNN or RNN `embedding_net`."""
+    if device == "cpu":
+        return "cpu"
+    else:
+        warnings.warn(
+            "GPU was selected as a device for training the neural network. "
+            "Note that we expect **no** significant speed ups in training for the "
+            "default architectures we provide. Using the GPU will be effective "
+            "only for large neural networks with operations that are fast on the "
+            "GPU, e.g., for a CNN or RNN `embedding_net`."
+        )
+        current_gpu_index = torch.cuda.current_device()
+        if device == "cuda":
+            return f"cuda:{current_gpu_index}"
+        else:
+            assert device == f"cuda:{current_gpu_index}", (
+                f"Unrecognized device {device}, "
+                "should be one of [`cpu`, `cuda`, f`cuda:{index}`]"
             )
-        except (RuntimeError, AssertionError):
-            warnings.warn(f"Device {device} not available, falling back to CPU.")
-            device = "cpu"
-
-    return device
+            return device
 
 
 def check_if_prior_on_device(device, prior: Optional[Any] = None):
     if prior is not None:
```

Comment:
general suggestion: we could have a 
```
if prior is None: 
    pass
else: 
    ...
```
to improve readability?

Reply:
Will change it thanks :)